lib/ansible/modules/systemd.py

```diff
@@ -487,9 +491,21 @@ def main():
                     (rc, out, err) = module.run_command("%s %s '%s'" % (systemctl, action, unit))
                     if rc != 0:
                         module.fail_json(msg="Unable to %s service %s: %s" % (action, unit, out + err))
-
                 result['enabled'] = not enabled
 
+            # Mask units after they are disabled
+            if not module.params['enabled'] and module.params['masked'] is not None:
+                if not module.check_mode and masked != module.params['masked']:
+                    (rc, out, err) = module.run_command("%s %s '%s'" % (systemctl, mask_action, unit))
+                    if rc != 0:
+                        # some versions of system CAN mask/unmask non existing services, we only fail on missing if they don't
+                        fail_if_missing(module, found, unit, msg='host')
+                    else:
+                        masked = module.params['masked']
+
+        if module.params['masked'] is not None:
+            result['masked'] = masked
+
```

Comment:
If we make `enabled` and `masked` mutually exclusive, this section of code is unnecessary.